### ISSUE TITLE
ToGateTrigger not triggering / how to delay a trigger

### ISSUE THREAD

**Issue by SimonStimberg** [npisanti/ofxPDSP#77]
Hi, first of all thanks for the amazing addOn!  
I am trying to use the ToGateTrigger to trigger an AHR envelope. The gate seems to work fine, as the `meter_gate()` outputs 1/0 at the correct timing. But the envelope is not getting triggered. Am I doing something wrong?

```
pdsp::LFOPhasor       impulseLFO;
pdsp::AHR             envA;
pdsp::AHR             envB;
pdsp::LHDelay         delaySignal;
pdsp::ToGateTrigger   gate;

// the patching is:

impulseLFO.set(0.5);    // send impulses at 0.5Hz
delaySignal.set(500);   // delay the signal by 500ms

impulseLFO.out_trig() >> envA;        // triggers envelope A
envA >> delaySignal >> gate >> envB;  // should trigger envelope B after 500ms
```

What I actually want to do is to have a cyclic impulse (the LFOphasor) to trigger envA and after a short delay trigger envB. As I want to have multiple instances from this module that are randomly instantiated with independent impulse rates, a sequencer grid seems not to be appropriate for this.  
Do you might have a hint to a solution for this?
Thanks in advance!

**Comment by npisanti** [comment]
the `AHR` envelope does not react to gate on/off as it doesnt have a sustain stage, maybe you want to do this: 
```
impulseLFO.set(0.5);    // send impulses at 0.5Hz
delaySignal.set(500);   // delay the signal by 500ms

impulseLFO.out_trig() >> envA;        // triggers envelope A
impulseLFO.out_trig() >> delaySignal >> envB;  // should trigger envelope B after 500ms
```
this way the `impulseLFO` will trigger both `envA` and `envB` (with delay) , without any need of `pdsp::ToGateTrigger `. You may also want to check out `pdsp::TriggerGeiger` that does what you do with an LFO but with an additional settable jitter.

Or are both `envA` and `envB` not being triggered?

**Comment by SimonStimberg** [comment]
thanks for the fast reply!  

unfortuanely ``impulseLFO.out_trig() >> delaySignal >> envB`` seems not to work. the envelope doesn't get triggered. but `envA` and `envB` both work fine when being triggered directly without a delay.  

I solved it so far as I am now delaying the resulting signal after `envB` - not the trigger itself.  
what is strange is that the delayed signal is being chopped off at the end when increasing the delay time. it seems like it would cut off the amount of delay time at the end of the signal. I already increased the delay's buffer size via `setMaxTime()` but this had no effect.
